SimpleClientDHTSSLWIFIMGR/network-selector.html

```diff
@@ -0,0 +1,484 @@
+      <footer>
+        <div class="dialog-buttons"><button class="secondary" onClick="cancelPasswordScreen(event);">Cancel</button><button class="submitButton" onClick="submitPasswordScreen(event);">OK</button></div>
+      </footer>
+    </div>
+  </div>
+  
+<script>
+
+var $ = document.querySelectorAll.bind(document);
+var $$ = document.querySelector.bind(document);
+
+var accessPointTpl = $$('#access-point');
+
+var activeSsid;
+
+function onClickSneezeguard(e) {
+  if (e.target == $$(".passwordScreen")) {
+    e.stopPropagation();
+    cancelPasswordScreen();
+  }
+}
+
+function cancelPasswordScreen() {
+  var passwordScreen = $$(".passwordScreen");
+  passwordScreen.querySelector(".passwordInput").value = "";
+  passwordScreen.classList.remove('enabled');
+}
+
+function submitPasswordScreen() {
+  // Called when user clicks OK.
+  var passwordScreen = $$(".passwordScreen");
+  var passwordInput = passwordScreen.querySelector(".passwordInput");
+  var password = passwordInput.value;
+  
+  if (password.length == 0) {
+    passwordInput.classList.add("jiggle");
+    setTimeout(()=> passwordInput.classList.remove("jiggle"), 500);
+    return;
+  }
+
+  submitAP(activeSsid, password);
+  
+  var submitButton = passwordScreen.querySelector(".submitButton");
+  submitButton.textContent = "Working...";
+  submitButton.disabled = true;
+  
+  setTimeout(() => {
+    submitButton.textContent = "OK";
+    submitButton.disabled = false;
+  }, 1000);
+
+}
+
+function submitAP(ssid, password) {
+  // This gets called when you click an unsecured network, or
+  // indirectly when you click OK after setting a password.
+  console.log("Submitting AP", ssid, password);
+}
+
+function onClickAP() {
+  console.log("Clicking", this, this.dataset);
+  var ssid = this.dataset.ssid;
+  var encrypted = this.dataset.encrypted;
+  if (encrypted) {
+    activeSsid = ssid; // store this globally for submitting later
+    var passwordScreen = $$(".passwordScreen");
+    passwordScreen.querySelector(".network-name").textContent = ssid;
+    passwordScreen.classList.add('enabled');
+    passwordScreen.querySelector(".passwordInput").focus();
+  } else {
+    submitAP(ssid);
+  }
+}
+
+function loadData(knownAPs) {
+  var container = $$('.access-points-container');
+  while (container.firstChild) {
+    container.firstChild.removeEventListener('click', onClickAP);
+    container.removeChild(container.firstChild);
+  }
+  
+  knownAPs.sort((a,b) => 
+    b.rssi - a.rssi
+  ).map(x => {
+    var rootElement = document.importNode(accessPointTpl.content, true).children[0];
+    
+    rootElement.querySelector('.access-point-ssid').textContent = x.ssid;
+    rootElement.querySelector('.access-point-encryption').classList.toggle('icon-lock', x.encryptionType.length)
+    var indicators = rootElement.querySelectorAll('.access-point-rssi-indicator');
+    indicators[2].classList.toggle('enabled', x.rssi >= 1);
```

Comment:
I figured it was silly to render rssi as a continuous value because it's such a coarse measure, but we could do a bar instead.